Sources/Common/Utils/StubVsImpl.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  StubVsImpl.swift
+//
+//
+//  Created by Anian Schleyer on 05.06.24.
+//
+
+import Foundation
+
+@propertyWrapper
+public struct StubOrImpl<Proto> {
```

Comment:
How about we conform factories to a protocol `DependencyFactory` instead, through which we can provide a default implementation:

```swift

protocol DependencyFactory {
    associatedtype Service

    static var shared: Service { get }
    static var live: Service { get }
    static var mock: Service { get }
}

extension DependencyFactory {
    static var shared: Service {
        CommandLine.arguments.contains("-Screenshots") ? Self.mock : Self.live
    }

    static var mock: Service {
        Self.live
    }
}

// MARK: Example

struct NumberFactory: DependencyFactory {
    static var live: Int {
        1
    }

    static var mock: Int {
        2
    }
}
```

This is similar to [swift-dependencies' approach](
https://github.com/pointfreeco/swift-dependencies/blob/76b98dea40d2b13ace67216cbb100cef673e8267/Sources/Dependencies/DependencyKey.swift#L156)